weave/workers/weave_scorer.py

```diff
@@ -264,56 +267,92 @@ async def process_monitor(
+                attributes={},
+            )
+        )
 
-    if isinstance(output, ObjectRef):
-        output = output.get()
+        call_start_res = self._server.call_start(call_start_req)
 
-    scorer_attributes = get_scorer_attributes(scorer)
-    score_op = scorer_attributes.score_op
-    score_signature = inspect.signature(score_op)
-    score_arg_names = list(score_signature.parameters.keys())
-    score_output_name = "output" if "output" in score_arg_names else "model_output"
-    score_arg_names = [param for param in score_arg_names if (param != "self")]
-    score_args = {k: v for k, v in example.items() if k in score_arg_names}
-    score_args[score_output_name] = output
+        return Call(
+            _op_name=op.name,
+            project_id=self._project_id,
+            parent_id=None,
+            trace_id=call_start_res.trace_id,
+            id=call_start_res.id,
+            inputs=inputs,
+        )
 
-    inputs_with_defaults = _default_on_input_handler(score_op, (), score_args).inputs
-    score_args = {**inputs_with_defaults, "self": scorer}
+    def finish_call(
+        self,
+        call: Call,
+        output: Any,
+        exception: BaseException | None = None,
+        *__: Any,
+        **___: Any,
+    ) -> None:
+        call_end_req = tsi.CallEndReq(
+            end=tsi.EndedCallSchemaForInsert(
+                project_id=self._project_id,
+                id=call.id,
+                ended_at=datetime.now(),
+                output=output,
+                summary={},
+            )
+        )
 
-    server = get_trace_server()
+        self._server.call_end(call_end_req)
 
-    call_start_req = tsi.CallStartReq(
-        start=tsi.StartedCallSchemaForInsert(
-            op_name=score_op.name,
-            project_id=project_id,
-            inputs=inputs_with_defaults,
-            started_at=datetime.now(),
-            attributes={},
-        )
-    )
 
-    call_start_res = server.call_start(call_start_req)
+async def _do_score_call(
+    scorer: Scorer, call: Call, project_id: str
+) -> tuple[str, Any]:
+    example = {k: v for k, v in call.inputs.items() if k != "self"}
+    output = call.output
 
-    result = score_op.resolve_fn(**score_args)
+    if isinstance(output, ObjectRef):
+        output = output.get()
 
-    result = box(result)
+    client = WeaveWorkerClient(project_id)
 
-    call_end_req = tsi.CallEndReq(
-        end=tsi.EndedCallSchemaForInsert(
-            project_id=project_id,
-            id=call_start_res.id,
-            ended_at=datetime.now(),
-            output=result,
-            summary={},
+    def _async_call_op(
+        op: Op, *args: Any, **kwargs: Any
+    ) -> Coroutine[Any, Any, tuple[Any, Call]]:
+        return asyncio.to_thread(
+            lambda: _call_sync_func(op, *args, client=client, **kwargs)
         )
+
+    apply_scorer_result = await apply_scorer_async(
+        scorer, example, output, _async_call_op
```

Comment:
This is where we reuse SDK logic instead of duplicating it. We need to customize `async_call_op` in order to inject our custom client as seen above.